app/main.py

```diff
@@ -1,44 +1,67 @@
+import asyncio
 import time
+from typing import Awaitable, Any
 
 from iot.devices import HueLightDevice, SmartSpeakerDevice, SmartToiletDevice
 from iot.message import Message, MessageType
 from iot.service import IOTService
 
 
-def main() -> None:
+async def run_sequence(*functions: Awaitable[Any]) -> None:
+    for function in functions:
+        await function
```

Comment:
In the `run_sequence` function, you're not actually calling the `function` in the loop. You're just awaiting the function object, which is a no-op. You probably meant to call the function, like this: `await function()`. But be careful, this only works if `function` is a coroutine function, not a coroutine object.